internal/metrics/metrics_test.go

```diff
@@ -0,0 +1,73 @@
+package metrics
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	dto "github.com/prometheus/client_model/go"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	numberOfRegisteredValue       = 3
+	numberOfFailedToRegisterValue =1
+	numberOfCreatedOBCsValue      =2
+)
+
+func TestMetrics(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Metrics Suite")
+}
+
+func findMetric(src []*dto.MetricFamily, query string) *dto.MetricFamily {
+	for _, s := range src {
+		if s.Name != nil && *s.Name == query {
+			return s
+		}
+	}
+	return nil
+}
+
+var _ = Describe("Metrics", func() {
+	m := New()
+
+	It("correctly passes calls to the collectors", func() {
+		//when
+		for  i := 0; i < numberOfRegisteredValue; i++ {
+			m.IncRegisteredEdgeDevice()
```

Comment:
Here three different methods are tested in one test cases; they should have their own, separate ones. The code as it is invites future developers adding more metrics to simply add their tests  here and after a while this one test might become a hundreds-of-lines monster. 

The verification part is also not so easy to understand at first glance.

As per [contributing guidelines](https://github.com/jakub-dzon/k4e-operator/blob/main/CONTRIBUTING.md#testing) any logic and conditionals in tests should be avoided.

I'm sorry for not pointing this out in my previous comment.

Reply:
I already tried to split it into 3 different test but I saw several issues - first the metrics object in the test cannot be initialized between the tests and also the return value from that line `data, err := metrics.Registry.Gather()` is a struct of all of the metrics so need to go over each one of them anyway

Reply:
Inability to have fresh metrics registry between tests is unfortunate, but it's not a big problem if the tests don't depend on each other.

This is test code and we should prioritize readability over performance/saving a few loops. The verification code: https://github.com/jakub-dzon/k4e-operator/pull/46/files#diff-8ea0d64df378dd743e9e7106592591a8568f325916af29bf4980054b6373b370R49 and https://github.com/jakub-dzon/k4e-operator/pull/46/files#diff-8ea0d64df378dd743e9e7106592591a8568f325916af29bf4980054b6373b370R62 is a bit cryptic.